jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/HiveQueryRewriter.java

```diff
@@ -35,6 +35,10 @@ public String rewriteColumnType(ColumnType columnType, Integer columnSize) {
         if (columnType == ColumnType.INTEGER) {
             return "INT";
         }
+        // Hive does not support VARCHAR without a width, nor VARCHAR(MAX).
+        if (columnType == ColumnType.STRING && columnSize == null) {
```

Comment:
Hmmm... It actually seems that Hive has a literal STRING type without length, that some actually recommends.

We've seen other cases of problems with STRING before (in SQL Server), but I can't remember the details. Do you remember if it is something to worry about, @kaspersorensen?